emulator/app/src/i3c_socket.rs

```diff
@@ -133,26 +139,250 @@ fn handle_i3c_socket_connection(
                 bus_command_tx.send(bus_command).unwrap();
             }
             Err(ref e) if e.kind() == ErrorKind::WouldBlock => {}
+            Err(ref e) if e.kind() == ErrorKind::ConnectionReset => {
+                println!("handle_i3c_socket_connection: Connection reset by client");
+                break;
+            }
             Err(e) => panic!("Error reading message from socket: {}", e),
         }
-        match bus_response_rx.try_recv() {
-            Ok(response) => {
-                let data_len = response.resp.data.len();
-                if data_len > 255 {
-                    panic!("Cannot write more than 255 bytes to socket");
+        if let Ok(response) = bus_response_rx.recv_timeout(Duration::from_millis(10)) {
+            let data_len = response.resp.data.len();
+            if data_len > 255 {
+                panic!("Cannot write more than 255 bytes to socket");
+            }
+            let outgoing_header = OutgoingHeader {
+                ibi: response.ibi.unwrap_or_default(),
+                from_addr: response.addr.into(),
+                response_descriptor: response.resp.resp,
+            };
+            let header_bytes: [u8; 6] = transmute!(outgoing_header);
+            stream.write_all(&header_bytes).unwrap();
+            stream.write_all(&response.resp.data).unwrap();
+        }
+    }
+}
+
+pub(crate) fn run_tests(
+    running: Arc<AtomicBool>,
+    port: u16,
+    target_addr: DynamicI3cAddress,
+    tests: Vec<Test>,
+) {
+    let running_clone = running.clone();
+    let addr = SocketAddr::from(([127, 0, 0, 1], port));
```

Comment:
This can be addressed in another PR, but since we are only communicating with the socket locally and not from another program, it might be easier to use the `mpsc` channels directly so that we don't have to worry about the ports. (Otherwise, we're going to run into issues where there is a port conflict since we hard code the port and don't check if it's already in use.)

Reply:
You are right! I did face a port conflict once while testing. I'll change to `mpsc` in the next PR when I add more tests.